codice/ddf
# DDF-4363 Allow admins to expose capabilities of sources through a new configuration

#### What does this PR do?
Allows admins to expose capabilities of sources through a new configuration. These capabilities will be available over the existing sources endpoint.
#### Who is reviewing it? 
@Bdthomson @hayleynorton @lamhuy 
#### Select relevant component teams: 
@codice/core-apis 
#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@bdeining
#### How should this be tested?
- build/run
- in admin > catalog create a few sources
- open the main UI and inspect for the "sources" network call
- verify that "capabilities" is an empty array for every source
- in admin > system > Source Capabilities, add a new config that matches a source id and anything can be entered for capabilities. save the config
- refresh the main UI and verify that the source now displays those capabilities in the network response
- add and delete more source capabilities for other sources in the same manner and verify no unexpected behavior

#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4363](https://codice.atlassian.net/browse/DDF-4363)
#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
